interfaces/cython/cantera/composite.py

```diff
@@ -676,12 +683,23 @@ def append(self, state=None, **kwargs):
             self._phase.state = state
 
         elif len(kwargs) == 1:
-            attr, value = next(iter(kwargs.items()))
+            attr, value = kwargs.popitem()
             if frozenset(attr) not in self._phase._full_states:
                 raise KeyError(
                     "'{}' does not specify a full thermodynamic state".format(attr)
                 )
-            setattr(self._phase, attr, value)
+            if normalize or attr[-1] == "Q":
+                setattr(self._phase, attr, value)
+            elif not normalize:
```

Comment:
I think this should just be `else`... What do you think @ischoegl, since you suggested `elif`?

Reply:
Yup. While `elif` works, the cases are mutually exclusive, so you’re correct; my main objection was against the `if` that was here before as it was possible to enter both clauses.